Demo/Tests/MoyaProviderSpec.swift

```diff
@@ -348,6 +348,65 @@ class MoyaProviderSpec: QuickSpec {
                 expect(executed).to(beTruthy())
             }
         }
+
+        describe("a provider with custom sample response closures") {
+            it("returns sample data") {
+                let endpointResolution: MoyaProvider<GitHub>.EndpointClosure = { target in
+                    let url = target.baseURL.appendingPathComponent(target.path).absoluteString
+                    return Endpoint(URL: url, sampleResponseClosure: {.networkResponse(200, target.sampleData)}, method: target.method, parameters: target.parameters)
+                }
+                let provider = MoyaProvider<GitHub>(endpointClosure: endpointResolution, stubClosure: MoyaProvider.ImmediatelyStub)
```

Comment:
Hmm, I see these are [class funcs](https://github.com/Moya/Moya/blob/198a23ff3cfcaa7fa34fd8014f33a18f1e180656/Source/Moya.swift#L122-L133). Why the uppercased functions?